components/camera/videosource/webcam.go

```diff
@@ -594,13 +596,52 @@ func (c *monitoredWebcam) NextPointCloud(ctx context.Context) (pointcloud.PointC
 	return c.exposedProjector.NextPointCloud(ctx)
 }
 
+func (c *monitoredWebcam) DriverInfo() (driver.Info, error) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	if c.underlyingSource == nil {
+		return driver.Info{}, errors.New("no configured camera")
+	}
+	d, err := gostream.DriverFromMediaSource[image.Image, prop.Video](c.underlyingSource)
+	if err != nil {
+		return driver.Info{}, errors.Wrap(err, "cannot get driver from media source")
+	}
+	return d.Info(), nil
+}
+
+//go:embed data/intrinsics.json
+var intrinsics []byte
+
 func (c *monitoredWebcam) Properties(ctx context.Context) (camera.Properties, error) {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 	if err := c.ensureActive(); err != nil {
 		return camera.Properties{}, err
 	}
-	return c.exposedProjector.Properties(ctx)
+
+	props, err := c.exposedProjector.Properties(ctx)
+	if err != nil {
+		return camera.Properties{}, err
+	}
+	if props.IntrinsicParams == nil {
+		dInfo, err := c.DriverInfo()
+		if err != nil {
+			return camera.Properties{}, errors.Wrap(err, "can't find driver info for camera")
+		}
+		var data map[string]transform.PinholeCameraIntrinsics
+		if err := json.Unmarshal(intrinsics, &data); err != nil {
+			return camera.Properties{}, errors.Wrap(err, "cannot parse intrinsics json")
+		}
+		cameraIntrinsics, exists := data[dInfo.Name]
+		if !exists {
+			c.logger.Info("Camera model not found in known camera models for: ", dInfo.Name, ". returning properties without intrinsics")
+			return props, nil
+		}
+		c.logger.Info("Intrinsics are known for camera model: ", dInfo.Name, ". adding intrinsics to camera properties")
+		props.IntrinsicParams = &cameraIntrinsics
```

Comment:
These logging messages will appear every time a user calls Properties, which might be frequent. I would say you should create a bool or something similar so that the log message gets written only once to the logs, and not on repeated Properties calls

Reply:
added this bool `hasLoggedIntrinsicsInfo` that is set back to `false` every time we hit `Reconfigure()`